app/inpututils.cpp

```diff
@@ -1842,57 +1842,84 @@ QgsGeometry InputUtils::createGeometryForLayer( QgsVectorLayer *layer )
     return geometry;
   }
 
-  switch ( layer->wkbType() )
+  bool isMulti = QgsWkbTypes::isMultiType( layer->wkbType() );
```

Comment:
Can we have a test for the `createGeometryForLayer` and `invalidGeometryWarning` functions? Thaks